src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,1346 @@
+        self._temperaturePar = None
+
+        self.temperatureTarget = model.FloatContinuous(0, unit="°C", range=(-273.15, 10**3),
+                                                       setter=self._setTemperatureTarget)
+        self.temperatureActual = model.FloatContinuous(0, unit="°C", range=(-273.15, 10**3), readonly=True)
+        self.temperatureRegulation = model.VAEnumerated(0, unit=None, choices={0: "off", 1: "on", 2: "rush"},
+                                                        setter=self._setTemperatureRegulation)
+        self.age = model.FloatContinuous(0, unit="h", readonly=True, range=(0, 10**9))
+        self.precursorType = model.StringVA("", readonly=True)
+
+        self.on_connect()
+
+    def on_connect(self):
+        """
+        Defines direct pointers to server components and connects parameter callbacks for the Orsay server.
+        Needs to be called after connection and reconnection to the server.
+        """
+        self._gis = self.parent.datamodel.HybridGIS
+        self._temperaturePar = self._gis.ReservoirTemperature
+
+        self._gis.ErrorState.Subscribe(self._updateErrorState)
+        self._gis.RodPosition.Subscribe(self._updateErrorState)
+        self._temperaturePar.Subscribe(self._updateTemperatureTarget)
+        self._temperaturePar.Subscribe(self._updateTemperatureActual)
+        self._gis.RegulationOn.Subscribe(self._updateTemperatureRegulation)
+        self._gis.RegulationRushOn.Subscribe(self._updateTemperatureRegulation)
+        self._gis.ReservoirLifeTime.Subscribe(self._updateAge)
+        self._gis.PrecursorType.Subscribe(self._updatePrecursorType)
+
+        self.update_VAs()
+
+    def update_VAs(self):
+        """
+        Update the VA's. Should be called after reconnection to the server
+        """
+        self._updateErrorState()
+        self._updateTemperatureTarget()
+        self._updateTemperatureActual()
+        self._updateTemperatureRegulation()
+        self._updateAge()
+        self._updatePrecursorType()
+
+    def _updateErrorState(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads the error state from the Orsay server and saves it in the state VA
+        """
+        if parameter not in (self._gis.ErrorState, self._gis.RodPosition, None):
+            raise ValueError("Incorrect parameter passed to _updateErrorState. Parameter should be "
+                             "datamodel.HybridGIS.ErrorState, datamodel.HybridGIS.RodPosition, or None. "
+                             "Parameter passed is %s." % parameter.Name)
+        if not attributeName == "Actual":
+            return
+
+        msg = ""
+        try:
+            rod_pos = int(self._gis.RodPosition.Actual)
+        except TypeError:
+            rod_pos = ROD_NOT_DETECTED
```

Comment:
Please log the value at the warning level. So that if for some reason it keeps saying it's not detected, we have an idea from the log what it sends as value.